rules-tests/DeadCode/Rector/FunctionLike/NarrowTooWideReturnTypeRector/Fixture/edge_cases.php.inc

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\FunctionLike\NarrowTooWideReturnTypeRector\Fixture;
+
+final class EdgeCases
+{
+    public function implicitNull(): string|int|null
+    {
+        if (rand(0, 1)) {
+            return;
+        }
+
+        return 'something';
+    }
+
+    public function implicitReturn(): string|int|null
+    {
+        if (rand(0, 1)) {
+            return 'something';
+        }
+    }
+
+    public function noReturn(): string|int|null
+    {
+        if (rand(0, 1)) {
+            echo 'something';
+        }
+    }
+
+    public function withMixed(): mixed|string|int
```

Comment:
This already error on very first usage, see https://3v4l.org/NS7GO

Again, Rector is not error handler :)